src/matrices.jl

```diff
@@ -296,10 +110,11 @@ recurrence quantifications", in: Webber, C.L. & N. Marwan (eds.), *Recurrence
 Quantification Analysis. Theory and Best Practices*, Springer, pp. 3-43 (2015).
 """
 function RecurrenceMatrix(x, ε; metric = DEFAULT_METRIC,
-    parallel::Bool = length(x) > 500 && Threads.nthreads() > 1, kwargs...)
+    parallel::Bool = length(x) > 500 && Threads.nthreads() > 1, fan::Bool = false,
+    kwargs...)
     m = getmetric(metric)
-    s = resolve_scale(x, m, ε; kwargs...)
-    m = recurrence_matrix(x, m, s, Val(parallel))
+    s = resolve_scale(x, m, ε; fan, kwargs...)
+    m = recurrence_matrix(x, m, s, Val(parallel), Val(fan))
     return RecurrenceMatrix(m)
 end
 
```

Comment:
@heliosdrm does your question target this function, where you said that `val` should be positional instead of keyword? If so, it makes no difference, since it is already positional in the `recurrence_matrix` call. Besides, you can never get rid of the type istability of going from `fan` to `Val(fan)` in every single function call. One function call will always remain type unstable, and it is best if this call is the highest level one.

Reply:
I mean in the constructor of `RecurrenceMatrix` itself. Now it is:

`RecurrenceMatrix(x, ε; kwargs...)`

There could be another method `RecurrenceMatrix(x, ε, fan::Val; kwargs...)`, such that if `fan isa `Val{false}` the result will contain a `Symmetric` sparse matrix, and if `fan isa `Val{ŧrue}` it will be a non-symmetric one. A bit less friendly than using `fan::Bool` in `kwargs`, as currently proposed. But we could have that method as a function barrier, available
together with the friendlier but type-unstable one.

Reply:
I like the idea of using the singleton types instead of `Val` types.